docs/checks/reference/ssh.md

```diff
@@ -5,13 +5,21 @@ SSH
 | ------------ | ------ | ------------ | ------------------------------------------------------ |
 | Host         | String | Y            | IP or FQDN of the host to run the SSH check against    |
 | Username     | String | Y            | The user to login with over SSH                        |
-| Password     | String | Y            | The password for the user that you wish to login with  |
+| Password     | String | Y (or PrivateKey)            | The password for the user that you wish to login with  |
+| PrivateKey   | String | Y (or Password)            | The absolute filepath of the private key file for the user that you wish to login with  |
```

Comment:
I would prefer to store the private key data itself instead of a file path. If a team needs to rotate keys, they won't be able to upload a new key file to wherever Dynamicbeat is running.

Copying and pasting the contents of a private key file into the Kibana UI _is_ pretty gross, but we don't have any way of supporting file uploads right now.

Reply:
Would you want to do base64 or some other form of encoding here, or just copy/paste the key into the check file? In testing I didn't see any of my checks load in Kibana, so if there is a way to modify the settings I haven't been able to get it to work for testing.

Reply:
Just copy/paste the key into the check file. Since it's JSON, all the newlines in the file would have to be replaced with `\n` characters.

Attributes can be modified in Kibana under the "scorestack" Kibana application. Teams can view and update the values of the user attributes for their checks - it's a basic web form with text input boxes. Pasting an SSH key into the text box wouldn't be pretty, but I think it's the easiest for teams to use and for us to implement.